txjuju/tests/test_cli.py

```diff
@@ -1,16 +1,278 @@
+        spec = BootstrapSpec("my-env", "lxd")
+        self.assertEqual(
+            repr(spec),
+            ("BootstrapSpec(name='my-env', driver='lxd', "
+             "default_series='trusty', admin_secret=None)"),
+            )
+
+    def test___eq___same_with_base_class(self):
+        """
+        BootstrapSpec == other is True when they are the same
+        and have the same class.
+        """
+        spec = BootstrapSpec("my-env", "lxd")
+        other = BootstrapSpec("my-env", "lxd")
+
+        self.assertTrue(spec == other)
+
+    def test___eq___same_with_sub_class(self):
+        """
+        BootstrapSpec == other is True when they are the same,
+        even if the other is a BootstrapSpec subclass.
+        """
+        spec = BootstrapSpec("my-env", "lxd")
+        other = type("SubSpec", (BootstrapSpec,), {})("my-env", "lxd")
+
+        self.assertTrue(spec == other)
+
+    def test___eq___same_with_other_class(self):
+        """
+        BootstrapSpec == other is True when they are the same,
+        even if the classes are different.
+        """
+        spec = BootstrapSpec("my-env", "lxd", "trusty", "pw")
+        other_cls = namedtuple("Sub",
+                               "name driver default_series admin_secret")
+        other = other_cls("my-env", "lxd", "trusty", "pw")
+
+        self.assertTrue(spec == other)
+
+    def test___eq___identity(self):
+        """
+        For BootstrapSpec, spec == spec is True.
+        """
+        spec = BootstrapSpec("my-env", "lxd")
+
+        self.assertTrue(spec == spec)
+
+    def test___eq___different(self):
+        """
+        BootstrapSpec == other is False when they differ.
+        """
+        spec = BootstrapSpec("my-env", "lxd")
+        other = BootstrapSpec("my-env", "spam")
+
+        self.assertFalse(spec == other)
+
+    def test___ne___same(self):
+        """
+        BootstrapSpec != other is False when they are the same.
+        """
+        spec = BootstrapSpec("my-env", "lxd")
+        other = BootstrapSpec("my-env", "lxd")
+
+        self.assertFalse(spec != other)
+
+    def test___ne___different(self):
+        """
+        BootstrapSpec != other is True when they differ.
+        """
+        spec = BootstrapSpec("my-env", "lxd")
+        other = BootstrapSpec("my-env", "spam")
+
+        self.assertTrue(spec != other)
+
+    def test_config(self):
+        """
+        BootstrapSpec.config() returns a Config containing
+        a ControllerConfig corresponding to the bootstrap spec.
+        """
+        spec = BootstrapSpec("my-env", "lxd", "xenial", "pw")
+        cfg = spec.config()
+
+        self.assertEqual(len(cfg.controllers), 1)
+        self.assertEqual(
+            cfg.controllers[0],
+            config.ControllerConfig(
+                "my-env",
+                config.CloudConfig("lxd", "lxd"),
+                config.BootstrapConfig("xenial", "pw"),
+                ),
+            )
+
+
+class TestAPIInfo(unittest.TestCase):
```

Comment:
Last nit and I'm done. test classes naming convention across the rest of txjuju carry the format <objectTested>Test. It makes things easier for dumb fgrep searching of related test classes for a given object.